### PR TITLE
some improvements for Makefile

### PR DESCRIPTION
**Pull request by undef21**
- add -Werror for SDL2 build;

- enable echoing for fheroes2 directory build. It's usefull for building linters and visual control.

related issue #2294

### CODE REVIEW

File: src/Makefile [hunk 1]
```diff
@@ -21,6 +21,7 @@ CXXFLAGS_TP := $(CXXFLAGS)
 CFLAGS := $(CXXFLAGS) -Wall -Wextra -Wpedantic
 
 ifdef WITH_SDL2
+CFLAGS := $(CFLAGS) -Werror
```
Comment (ihhub) [inline src/Makefile:24]: Please declare an extra environment variable called `FHEROES2_STRICT_COMPILATION`. Once it's set we add this compilation warning.
Please don't forget to update makefile in the root to include a description of the new variable.

Then later please change these files:
```
.github/workflows/osx_pull_request.yml
.github/workflows/osx_release.yml
```
To add this new variable like it's done for `WITH_SDL2: "ON"`.

On top of this please modify **.travis.yml** file to add changes in 2 builds like `- export WITH_SDL2="ON"`.

File: src/dist/Makefile [hunk 2]
```diff
@@ -59,8 +59,7 @@ $(RES): $(ICOFILE)
 VPATH := $(SRCDIRLIST)
 
 %.o: %.cpp
-	@echo "cxx: $@"
-	@$(CXX) -c -MD $(addprefix -I, $(SRCDIRLIST)) $< $(CFLAGS)
+	$(CXX) -c -MD $(addprefix -I, $(SRCDIRLIST)) $< $(CFLAGS)
```
Comment (ihhub) [inline src/dist/Makefile:62]: Please revert this change for now as it's not related to the issue. We can do this separately.

**Review by ihhub — CHANGES_REQUESTED** [review]
Hi @undef21 I describe steps which we need to do. Please take a look.

File: src/Makefile (hunk 1 above)
Comment (undef21) [reply src/Makefile:24]: Done.

File: src/dist/Makefile (hunk 2 above)
Comment (undef21) [reply src/dist/Makefile]: Done.

**Comment by ihhub** [comment]
@undef21 thank you so much for this improvement!

**Pull request merged**